tests/python/unittest/test_tir_transform_compact_buffer_region.py

```diff
@@ -505,6 +505,74 @@ def compacted_opaque_access_annotated_func(a: T.handle) -> None:
                 T.store(C.data, i, T.load("float32", B.data, i))
 
 
+@T.prim_func
+def sparse_read_cache(
+    A_data: T.Buffer[(819,), "float32"],
+    B: T.Buffer[(128,), "float32"],
+    A_indptr: T.Buffer[(129,), "int32"],
+    A_indices: T.Buffer[(819,), "int32"],
+) -> None:
+    for i in T.serial(128):
+        with T.block("rowsum_outer"):
+            T.reads(
+                A_indptr[i],
+                A_indptr[i + 1],
+                A_data[A_indptr[i] + 0 : A_indptr[i] + (A_indptr[i + 1] - A_indptr[i])],
```

Comment:
```suggestion
                A_data[A_indptr[i] : A_indptr[i + 1]],
```

Reply:
Due to some parser issue we're now allowed to simplify the expression for now:
```
error: Unsupported index type, expected int or tvm.tir.PrimExpr, but got <class 'tvm.script.tir.node.BufferSlice'>
 --> tensorir.py:431:44
     |  
 431 |              T.reads(A_indptr[vi : vi + 1], A_data[A_indptr[vi]: A_indptr[vi + 1]])
     |                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 
note: run with `TVM_BACKTRACE=1` environment variable to display a backtrace.
```

I suppose this is some bug that can be easily fixed. However, I didn't get enough time to take a deeper look into the parser.

Reply:
One problem with this analysis is that we cannot assume `A_indptr[i+1] >= A_indptr[i]`.

Reply:
Good catch! I never thought of this before. Is it okay by specifying `assume_in_range(A_indptr[i + 1] - A_indptr[i], [0, 127])` in the future? (not pretty sure)

Reply:
Yes I think so, but we merge this one first.